src/Symfony/Component/Translation/Translator.php

```diff
@@ -491,4 +492,19 @@ protected function assertValidLocale($locale)
             throw new \InvalidArgumentException(sprintf('Invalid "%s" locale.', $locale));
         }
     }
+
+    /**
+     * @see WarmableInterface
+     *
+     * @param string $cacheDir only used for interface compatibility, $this->cacheDir is already defined
+     */
+    public function warmUp($cacheDir)
+    {
+        if ($this->locale !== null) {
```

Comment:
I did it in a first personnal draft, but if I check ```$this->cacheDir```, I have to define the 'annotations' subfolder (because $cacheDir is basically ```%kernel.cache_dir%```) and I thought it was not a very good idea to have some "configuration" info directly in my implementation. Do you think about another way to do this, without doing a dirty ```$cacheDir . '/annotations'``` ?
Perhaps ```TranslationsCacheWarmer::warmup``` could that. Would that be cleaner ?